From sam@ravnborg.org  Mon Mar  5 10:30:53 2007
From: Sam Ravnborg <sam@ravnborg.org>
Date: Sat, 3 Mar 2007 20:09:41 +0100
Subject: usb-storage: do not rebuild when kernel version changes
To: <usb-storage@lists.one-eyed-alien.net>, Greg KH <greg@kroah.com>, Andrew Morton <akpm@osdl.org>
Message-ID: <20070303190941.GB30463@uranus.ravnborg.org>
Content-Disposition: inline


Replacing use of UTS_RELEASE with utsname()->release
avoids that the usb-storage driver is recompiled each
time the kernel version changes.

Signed-off-by: Sam Ravnborg <sam@ravnborg.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/storage/usb.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- gregkh-2.6.orig/drivers/usb/storage/usb.c
+++ gregkh-2.6/drivers/usb/storage/usb.c
@@ -55,7 +55,7 @@
 #include <linux/slab.h>
 #include <linux/kthread.h>
 #include <linux/mutex.h>
-#include <linux/utsrelease.h>
+#include <linux/utsname.h>
 
 #include <scsi/scsi.h>
 #include <scsi/scsi_cmnd.h>
@@ -547,7 +547,7 @@ static int get_device_info(struct us_dat
 				idesc->bInterfaceSubClass,
 				idesc->bInterfaceProtocol,
 				msgs[msg],
-				UTS_RELEASE);
+				utsname()->release);
 	}
 
 	return 0;
